src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+---
+title: Promises
+layout: default
+category: Infusion
+---
+
+# Infusion Promises API
+
+[Promises](https://en.wikipedia.org/wiki/Futures_and_promises) are a now widespread programming construct aiming to 
+simplify coding of complex workflows involving values which may be available asynchronously (perhaps as a result
+of requiring I/O) or fallibly. JavaScript enjoys numerous competing libraries implementing this feature, such as 
+[when.js](https://github.com/cujojs/when), [Q](http://documentup.com/kriskowal/q) and [Bluebird](https://github.com/petkaantonov/bluebird)
+as well as even multiple competing promise standards, such as [Promises/A+](https://promisesaplus.com/) and others from [CommonJS](http://wiki.commonjs.org/wiki/Promises).
+Promises are even built into an upcoming version of the JavaScript language itself, [ES6](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+
+Infusion required an extremely simple implementation which can evolve independently, as well as begin the process of merging with
+the general declarative facilities of [Infusion IoC](HowToUseInfusionIoC.md) and hence become invisible as code. The implementation described here is a 
+transitional, mostly procedural system on its way to being summarised as configuration - as numerous Infusion features have
+been in their turn over the years.
+
+## Notes on interoperability
+
+As commented [below](#commentary-on-requirements), Infusion promises have taken a different set of tradeoffs to many of those elsewhere in the industry. In terms of interoperability,
+Infusion promises are at least universally recognised as a "[foreign thenable](https://github.com/cujojs/when/blob/master/docs/api.md#when)" and hence can be easily adapted into promises of any of the other libraries.
+In terms of promise algorithms, since Infusion promises meet a weaker contract than usual, Infusion promises cannot safely be supplied to the promise
+algorithms of other libraries without adaptation. However, promises from foreign libraries can easily be used within Infusion's algorithms.
+
+## Core Promises API
+
+### fluid.promise()
+
+* Returns: `{Promise}`
```

Comment:
The structure of an Infusion "Promise" object should be documented somewhere. If it already is, this should link to that.